cmake/CeleritasLibrary.cmake

```
@@ -92,11 +168,11 @@ function(celeritas_sources_contains_cuda var)
   set(_result)
   foreach(_source ${ARGN})
     get_source_file_property(_iscudafile "${_source}" LANGUAGE)
-    if(_iscudafile STREQUAL "CUDA")
+    if ("x${_iscudafile}" STREQUAL "xCUDA")
       list(APPEND _result "${_source}")
     elseif(NOT _iscudafile)
       get_filename_component(_ext "${_source}" LAST_EXT)
-      if(_ext STREQUAL ".cu")
+      if("${_ext}" STREQUAL ".cu")
```

```suggestion
      if(_ext STREQUAL ".cu")
```
and a bunch others in this file...